### PR TITLE
lua: Add mtev.inet_pton() function

### PR DESCRIPTION
**Pull request by HeinrichHartmann**
Apparently inet_pton is the best way to validate IP v4/v6 address: https://stackoverflow.com/a/4017219/1209380

Avoids ugly regexpes.

### CODE REVIEW

**Comment by HeinrichHartmann** [comment]
I should have added a comment about this: This function can be trivially extended to also return the `struct in_addr` if there is ever the need for it.
So it would look like:
```
rc, fam, addr = mtev.inet_pton(addr_str)
```

**Comment by postwait** [comment]
If it is trivial then extend, if it is not, then rename it.

**Comment by HeinrichHartmann** [comment]
updated

File: src/modules/lua_mtev.c [hunk 1]
```diff
@@ -5513,6 +5513,44 @@ nl_getaddrinfo(lua_State *L) {
...
+    lua_pushstring(L, "inet");
+    struct in_addr *u = lua_newuserdata(L, sizeof(struct in_addr));
+    *u = addr;
+    return 3;
+  }
+  rc = inet_pton(AF_INET6, host, &addr);
+  if (rc == 1) {
+    lua_pushboolean(L, 1);
+    lua_pushstring(L, "inet6");
+    struct in_addr *u = lua_newuserdata(L, sizeof(struct in_addr));
```
Comment (postwait) [inline src/modules/lua_mtev.c]: this isn't an in_addr... it's an `struct in6_addr`

Comment (HeinrichHartmann) [reply src/modules/lua_mtev.c]: Uh. The manual was not exactly transparent about this:

>     The inet_pton() function converts a presentation format address (that is,
>     printable form as held in a character string) to network format (**usually
>     a struct in_addr** or some other internal binary representation, in network
>     byte order).

Where is the return type documented?

**Pull request merged**